client/js/zip/z-worker.js

```diff
@@ -0,0 +1,153 @@
+/* jshint worker:true */
+(function main(global) {
```

Comment:
add a quick comment at the top saying `/* taken from / part of zip.js library ` so its clear? You have a good block comment in the `zip.js` file, so doing the same here would be great.